lib/src/kbasesearchengine/common/GUID.java

```diff
@@ -128,21 +152,54 @@ public GUID(String textGUID) {
             this.subObjectId = subObjectPart.substring(slashPos + 1);
         }
     }
-    
-    //TODO NOW if the version or accessGroupID is missing, how do you know which is which? Need to know storage code -> has access group mapping or have a placeholder (-1) for missing access group
+
+    /**
+     * Returns the URL-encoded version of this GUID.
+     *
+     * For the sake of readability the URL-encoded format of the GUID is,
+     *
+     *   {storageCode}___[{AccessGroupID}_]{accessGroupObjectId}[_{version}][___{subObjectType}_{subObjectId}],
+     *
+     *  where special characters in the individual tokens will get URL-encoded.
+     *
+     * @return URL-encoded version of this GUID
+     */
+    public String getURLEncoded() throws IOException {
+
+        String encodedGUID = toString("___", "_");
+
+        try {
+
+             encodedGUID =  URLEncoder.encode(encodedGUID, "UTF-8");
+
+        } catch(UnsupportedEncodingException ex ) {
+            // should never occur since the encoding is set to UTF-8 which is supported
+            throw new IOException(ex.getMessage());
```

Comment:
I'd throw a RuntimeException in that case. No need to make the user deal with an exception they'll never see.

Reply:
Done.